adapter/src/dummy.rs

```diff
@@ -30,16 +33,28 @@ impl DummyAdapter {
     }
 }
 
+#[derive(Debug)]
+pub struct Error {}
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Dummy Adapter error occurred!")
+    }
+}
+
+impl AdapterErrorKind for Error {}
+
 impl Adapter for DummyAdapter {
-    fn unlock(&mut self) -> AdapterResult<()> {
+    type AdapterError = Error;
```

Comment:
We don't need different errors for Ethereum/Dummy Adapter

Reply:
For our use case sharing an Error struct between `Ethereum/Dummy` Adapter works fine. Also, if we add additional adapters in the future easily reusable and extensible

Reply:
Not every error apply for every adapter. Dummy adapter doesn't use Keystore, why should you have a Keystore error? It's not a general error.

Reply:
You are thinking in terms Dummy/Ethereum Adapter.  The DummyAdapter is just for development (it doesn't warrant the added complexity), if possible it shouldn't be compiled for release. When/If we add another adapter it's most likely going to use the same errors with the Ethereum Adapter.

Reply:
I think in terms of `trait Adapter` you cannot generalize every single error an implementation of the trait can have in a single `Error` since every trait implementation can have different impl Adapter specific errors.

Reply:
There might be other ways to do it. Think of std::io::Error for example - you have the Error, but also the ErrorKind. I made it slightly different - leaving the non-specific Adapter Errors in the `adapter::Error` and then having a trait `AdapterErrorKind` to be implemented by each adapter specific Error.
_This can be changed later on._

https://doc.rust-lang.org/std/io/struct.Error.html

Reply:
There is no issue with AdapterErrorKind.

The point is Dummy Adapter throws every error Ethereum Adapter has. The only reason why it's not visible is because of the oversimplification of signatures to formatted strings and ignoring a number of cases because "development" etc. 

There is no point in having separate error struct for both of them. They both should throw the same errors. 

The Ethereum Adapter Error struct should also be used for Dummy Adapter.  In fact, we can model `Keystore error` for `DummyAdapter`  as a case where identity passed doesn't match the identity in `prep-db` because "technically" that is it  Keystore. 

The error struct should be shared, different error structs doesn't really offer any benefits. 
It leads having to duplicate the same errors in ETH to Dummy. That's why in the code for Dummy Adapter https://github.com/AdExNetwork/adex-validator-stack-rust/blob/cfc8c6d73b25158e8b3cb31aa20b49b0d160bba7/adapter/src/dummy.rs#L40 there is no implementation there because it would require duplicating error types.